### ISSUE TITLE
Connecting to namespaces hanging

### ISSUE THREAD

**Issue by bbrincat** [invisibleroads/socketIO-client#117]
Hi,

I've come across what I think is a bug. I'm working with Flask-SocketIO v2 as a server.
I'm trying to connect to multiple namespaces, in the same manner as described in the docs.

```
with SocketIO('localhost', 8000, LoggingNamespace) as socketIO:
    events_namespace = socketIO.define(LoggingNamespace, '/events')
    alarms_namespace = socketIO.define(LoggingNamespace, '/alarms')
    socketIO.wait()
```

however this is hanging when it's in `socketIO.define(LoggingNamespace, '/events')`

The problem seems to be in
```
    def define(self, Namespace, path=''):
        self._namespace_by_path[path] = namespace = Namespace(self, path)
        if path:
            self.connect(path)
            self.wait(for_connect=True)
        return namespace
```

On initialising, it implicitly connects to the root namespace, and doesn't wait for CONNECT packet from the server, ( and the server doesn't send the CONNECT packet).

On defining the second namespace, it waits for the packet to be send from the server. If the Flask-SocketIO  the connect returns True , then the CONNECT packet is received by the client.
```
@socketio.on('connect', namespace='/events')
def connect_events(*args):
    return True
```

However to break out of the wait loop: the following is being checked. Both namespaces ('/' and '/events') are checked for `_connected`. For '/' this is never set to True, because 1) it doesn't listen 2) server doesn't send 3) the SocketIO spec doesn't seem to specify it should send.

```
    def _should_stop_waiting(self, for_connect=False, for_callbacks=False):
        if for_connect:
            for namespace in self._namespace_by_path.values():
                is_namespace_connected = getattr(
                    namespace, '_connected', False)
                if not is_namespace_connected
                    return False
            return True
....
```

excluding the root namespace in the following way  fixes the issue, since the path is an empty string

    ` if not is_namespace_connected and namespace.path:`

Is this a bug or is there some way this should be done properly?

I've come across someone asking on StackOverflow encountering the same issue as well, with no solution.

Thanks, Ben

**Comment by invisibleroads** [comment]
The client will hang when defining a namespace if the namespace is not defined on the server.

What do you see when you enable logging?

```python
import logging
logging.getLogger('requests').setLevel(logging.WARNING)
logging.basicConfig(level=logging.DEBUG)
```

```
DEBUG:root:localhost:8880/socket.io [engine.io transport selected] websocket
DEBUG:root:localhost:8880/socket.io [engine.io heartbeat reset]
DEBUG:root:localhost:8880/socket.io [socket.io packet sent] 0/q,
DEBUG:root:localhost:8880/socket.io [socket.io waiting for connection]
DEBUG:root:localhost:8880/socket.io [socket.io packet received] 0
DEBUG:root:localhost:8880/socket.io [socket.io connected]
DEBUG:root:localhost:8880/socket.io /q [socket.io waiting for connection]
DEBUG:root:localhost:8880/socket.io [socket.io packet received] 4/q,"Invalid namespace"
WARNING:root:localhost:8880/socket.io /q [socket.io error] Invalid namespace
DEBUG:root:localhost:8880/socket.io /q [socket.io waiting for connection]
DEBUG:root:localhost:8880/socket.io /q [socket.io waiting for connection]
DEBUG:root:localhost:8880/socket.io /q [socket.io waiting for connection]
```

#115

**Comment by invisibleroads** [comment]
Duplicate of https://github.com/invisibleroads/socketIO-client/issues/84

Fixed in 2abcc91fb7f6c5a5d1da1f5edf2fda1d85890015

Please re-open this issue if it is still a problem.
